crates/oct-cloud/src/aws.rs

```diff
@@ -396,9 +356,55 @@ impl Resource for Ec2Instance {
             .ok_or("No instances returned")?;
 
         self.id = instance.instance_id.clone();
-        self.arn = instance.outpost_arn.clone();
-        self.public_ip = instance.public_ip_address.clone();
-        self.public_dns = instance.public_dns_name.clone();
+
+        log::info!("Instance launched, waiting for metadata to be available");
+
+        tokio::time::sleep(std::time::Duration::from_secs(30)).await;
+
+        // Poll for metadata
+        let mut attempts = 0;
+        const MAX_ATTEMPTS: usize = 10;
+        const SLEEP_DURATION: std::time::Duration = std::time::Duration::from_secs(5);
+
+        while attempts < MAX_ATTEMPTS {
+            if let Some(instance_id) = &self.id {
+                if let Ok(describe_response) =
+                    self.client.describe_instances(instance_id.clone()).await
+                {
+                    if let Some(reservation) = describe_response.reservations().get(0) {
+                        if let Some(instance) = reservation.instances().get(0) {
+                            // Update metadata fields
+                            if let Some(public_ip) = instance.public_ip_address() {
+                                self.public_ip = Some(public_ip.to_string());
+                            }
+                            if let Some(public_dns) = instance.public_dns_name() {
+                                self.public_dns = Some(public_dns.to_string());
+                            }
+
+                            // Break if all metadata is available
+                            if self.public_ip.is_some() && self.public_dns.is_some() {
+                                break;
+                            }
```

Comment:
Is it possible to have only one of them not None? I suppose if one of them is available, another one should also be available